internal/translator/utils/template_functions.go

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"regexp"
+	"strings"
+)
+
+// StartWithVowel validates if a word starts with vowel
+func StartWithVowel(data string) bool {
+	validPattern := regexp.MustCompile(`^[aeiouAEIOU]+`)
+
+	return validPattern.MatchString(data)
+}
+
+// StartWithConsonant validates if a word starts with consonant
+func StartWithConsonant(data string) bool {
+	validPattern := regexp.MustCompile(`^[aeiouAEIOU]+`)
```

Comment:
I think you can add flags to the regex to don't validate the case. It means case insensitive.